includes/migrate.php

```diff
@@ -23,6 +23,15 @@ function maybe_migrate() {
 		return;
 	}
 
+	// Schedule additional migrations if this is a multisite network.
+	if ( is_multisite() && is_super_admin() && ! get_site_option( 'wp101-bulk-migration-lock', false ) ) {
+		if ( ! wp_next_scheduled( 'wp101-bulk-migration' ) ) {
+			wp_schedule_single_event( time(), 'wp101-bulk-migration' );
+		}
+
+		add_site_option( 'wp101-bulk-migration-lock', true );
```

Comment:
If there is an error with the migration, most likely with `$key = $api->exchange_api_key();`, is there any way to clear this option so the admin would be able to initiate this process again without reaching directly into the `wp_options` table?

Reply:
Good call — I've updated this so that if an error occurs the lock is removed automatically. It can produce a little bit of overhead if we're already 95% of the way through the migration, but the `api_key_needs_migration()` check ensures we're not hitting the API to exchange an already-valid key.